schemas/theme/section.json

```diff
@@ -4,231 +4,193 @@
       "minimum": 1,
       "maximum": 2
     },
+
     "settings": {
       "$ref": "./settings.json"
     },
+
     "max_blocks": {
       "type": "integer",
-      "description": "The maximum number of blocks allowed in the section.",
+      "description": "There's a limit of 50 blocks per section. You can specify a lower limit with the max_blocks attribute",
+      "markdownDescription": "There's a limit of 50 blocks per section. You can specify a lower limit with the `max_blocks` attribute.\n\n---\n\n[Shopify reference](https://shopify.dev/docs/themes/architecture/sections/section-schema#max_blocks)",
       "minimum": 1,
       "maximum": 50
     },
+
     "blocks": {
       "type": "array",
-      "description": "Section blocks.",
+      "description": "Blocks are reusable modules of content that can be added, removed, and reordered within a section.",
+      "markdownDescription": "Blocks are reusable modules of content that can be added, removed, and reordered within a section.\n\n---\n\n[Shopify reference](https://shopify.dev/docs/themes/architecture/sections/section-schema#blocks)",
+      "properties": {
+        "type": {
+          "description": "The block type. Can be one of the following values: @theme, @app, or a custom block type.",
+          "markdownDescription": "The block type. Can be one of the following values: @theme, @app, or a custom block type.\n\n---\n\n[Shopify reference](https://shopify.dev/docs/themes/architecture/sections/section-schema#blocks)"
+        }
+      },
       "items": {
         "type": "object",
+        "required": ["type"],
         "properties": {
-          "name": {
-            "type": "string",
-            "description": "The block name."
-          },
           "type": {
             "type": "string",
-            "description": "The block type."
-          },
-          "settings": {
-            "$ref": "./settings.json"
+            "description": "The type of block that can be added to this block.",
+            "markdownDescription": "The type of block that can be added to this block.\n\n---\n\n[Shopify reference](https://shopify.dev/docs/themes/architecture/blocks/theme-blocks/schema#blocks)"
           }
         },
-        "if": {
-          "properties": {
-            "type": {
-              "const": "@app"
+        "$comment": "The allOf rule here exists because that's how we do discriminated unions in JSON schemas. If a rule matches, that rule will be used to document the type property. Otherwise we fallback to the docs above.",
+        "allOf": [
+          {
+            "if": { "required": ["type"], "properties": { "type": { "const": "@theme" } } },
+            "then": { "$ref": "./theme_block_entry.json" }
+          },
+          {
+            "if": { "required": ["type"], "properties": { "type": { "const": "@app" } } },
+            "then": { "$ref": "./app_block_entry.json" }
```

Comment:
Those if, required, properties things here is what enables us to do "discriminated" unions and show the relevant docs depending on the type value. 

`@app`? -> show app block docs
`@theme`? -> show theme blocks docs
`string` with `name`? -> show local block docs
`string` without `name`? -> show specific theme block docs